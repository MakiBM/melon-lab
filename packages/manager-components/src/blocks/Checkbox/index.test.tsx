import React from 'react';
import Checkbox from './index';

describe('Checkbox', () => {
  const defaultElement = (
    <Checkbox
      name={'Default name'}
      value={'Default value'}
      text={'Default text'}
    />
  );

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be selectable by class "checkbox"', () => {
    expect(shallow(defaultElement).is('.checkbox')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(defaultElement).find('.checkbox').length).toBe(1);
  });

  it('should render to static HTML', () => {
    expect(render(defaultElement).text()).toEqual('Default text');
  });
});
