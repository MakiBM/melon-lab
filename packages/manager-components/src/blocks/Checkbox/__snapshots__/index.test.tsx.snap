// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
<label
  className="jsx-__scopedHash checkbox"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <input
    className="jsx-__scopedHash checkbox__input"
    name="Default name"
    type="checkbox"
    value="Default value"
  />
  <span
    className="jsx-__scopedHash checkbox__checkmark"
  />
  <span
    className="jsx-__scopedHash checkbox__text"
  >
    Default text
  </span>
</label>
`;
